### PR TITLE
Update identityconnecto docs

### PR DESCRIPTION
**Pull request by pubudug**

### CODE REVIEW

File: docs/meshstack.customer-group-sync.md [hunk 1]
```diff
@@ -3,142 +3,192 @@ id: meshstack.customer-group-sync
...
+The synchronization process is done in three parts.
 
-<!--snippet:mesh.identityconnector.ldap.core#type-->
+* Collect
+* Transform
+* Transport
+
+### Collect
+
+In the Collect step, we connect to the LDAP server and fetch the LDAP entities. In order to do that,
```
Comment (chewbeccy) [inline docs/meshstack.customer-group-sync.md:20]:
```suggestion
In the "Collect" step, we connect to the LDAP server and fetch the LDAP entities. In order to do that,
```
```suggestion
In the Collect step, we connect to the LDAP server and fetch the LDAP entities. In order to do that,
```

File: docs/meshstack.customer-group-sync.md [hunk 2]
```diff
@@ -3,142 +3,192 @@ id: meshstack.customer-group-sync
...
 
 <!--DOCUSAURUS_CODE_TABS-->
 <!--Dhall Type-->
 ```dhall
-let ConnectionConfiguration =
+let SourceConfiguration =
     {-
-      url: The LDAP connection URL which would be of the form ldap://example.com:389
+      url:
+          The LDAP connection URL, including the ldap protocol
```
Comment (chewbeccy) [inline docs/meshstack.customer-group-sync.md:33]: why did you get rid of the examples?

File: docs/meshstack.customer-group-sync.md [hunk 3]
```diff
@@ -3,142 +3,192 @@ id: meshstack.customer-group-sync
...
-    mesh-object:
-      The type of meshObject this configuration is about. Can be one of
-      "MeshUser", "MeshCustomerUserGroup" or "MeshCustomerGroupBinding"
-    fields:
-      A list of configuration objects specifying which LDAP attribute gets mapped to which
-      meshObject attribute, and whether any transformations should be done on the attribute before
-      assigning it to the meshObject field.
-    tags: Same as fields above, but for tags on the specified meshObject.
+      An AttributeTransformation is used to define an attribute of an LDAP entry and how to transform it so that it
+      can be assigned to a field in a meshObject. There are two types of AttributeTransformations, Static and Regex.
```
Comment (chewbeccy) [inline docs/meshstack.customer-group-sync.md:131]: can you provide an example like group name?

File: docs/meshstack.customer-group-sync.md [hunk 4]
```diff
@@ -3,142 +3,192 @@ id: meshstack.customer-group-sync
...
+     until a match is found. If a match is found and the rule defines a value, the value will be used. If a value is not
+     set, the first matching group in the regular expression is used and the optional template is applied. If no
+     regular expressions match, the otherwise value will be assigned to the meshObject field.
+
       rules:
           A list of regex rules that the LDAP attributes will be tested against for a match.
           The matching is performed sequentially until a match is found.
+          A single rule has the following structure
+          value:
+              This is an optional parameter. If it is defined, if the LDAP attribute matches the
```
Comment (chewbeccy) [inline docs/meshstack.customer-group-sync.md:186]:
```suggestion
              This is an optional parameter. It is defined, if the LDAP attribute matches the
```
```suggestion
              This is an optional parameter. If it is defined, if the LDAP attribute matches the
```

File: docs/meshstack.customer-group-sync.md [hunk 5]
```diff
@@ -3,142 +3,192 @@ id: meshstack.customer-group-sync
...
+              regular expression, this value will be assigned as the value of the meshObject field.
+          regex:
+              The regular expression against which the LDAP attribute should be matched. If the value parameter
```
Comment (chewbeccy) [inline docs/meshstack.customer-group-sync.md:189]:
```suggestion
              The regular expression against which the LDAP attribute is compared?. If the value parameter
```
```suggestion
              The regular expression against which the LDAP attribute should be matched. If the value parameter
```

File: docs/meshstack.customer-group-sync.md (hunk 2 above)
Comment (pubudug) [reply docs/meshstack.customer-group-sync.md:33]: The example is moved to it's own dhall example section below.

File: docs/meshstack.customer-group-sync.md (hunk 3 above)
Comment (pubudug) [reply docs/meshstack.customer-group-sync.md:131]: A complete example are given in the dhall example sections below.

File: docs/meshstack.customer-group-sync.md (hunk 5 above)
Comment (pubudug) [reply docs/meshstack.customer-group-sync.md:189]: When it comes to regular expressions, `match` is the word that is usually used.

File: docs/meshstack.customer-group-sync.md (hunk 4 above)
Comment (pubudug) [reply docs/meshstack.customer-group-sync.md]: Changed the wording to `If it is defined and if  the LDAP ....`

File: docs/meshstack.customer-group-sync.md (hunk 3 above)
Comment (JohannesRudolph) [reply docs/meshstack.customer-group-sync.md:131]: @pubudug I'm for keeping examples in the code tabs for individual types (there's a Type + Example tab)

It's great if you give a full end to end example for identity connector configuration, however to me that's more of a "how to" than a reference documentation

**Review by chewbeccy — APPROVED** [review]

File: docs/meshstack.customer-group-sync.md (hunk 3 above)
Comment (pubudug) [reply docs/meshstack.customer-group-sync.md:131]: Yeah. This PR is organized so that the examples show in it's own tab.

**Comment by pubudug** [comment]
adding automerge label because the snippets not being available is blocking other doc updates.

**Pull request merged**